pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -0,0 +1,369 @@
+// limitations under the License.
+
+package supportbundle
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+	fakerest "k8s.io/client-go/rest/fake"
+	cgtesting "k8s.io/client-go/testing"
+
+	"antrea.io/antrea/pkg/apis/crd/v1beta1"
+	systemv1beta1 "antrea.io/antrea/pkg/apis/system/v1beta1"
+	antrea "antrea.io/antrea/pkg/client/clientset/versioned"
+	fakeclientset "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	"antrea.io/antrea/pkg/client/clientset/versioned/scheme"
+)
+
+var (
+	clientConfig = &rest.Config{
+		APIPath: "/supportbundle",
+		ContentConfig: rest.ContentConfig{
+			NegotiatedSerializer: scheme.Codecs,
+			GroupVersion:         &schema.GroupVersion{Group: "system.antrea.io", Version: "v1beta1"},
+		},
+	}
+	controllerInfo = v1beta1.AntreaControllerInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-controller",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-1",
+		},
+	}
+	node1 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "node-1",
+			ResourceVersion: "0",
+		},
+		Status: v1.NodeStatus{
+			Addresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "172.16.0.0",
+				},
+			},
+		},
+	}
+	node2 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node-1",
+		},
+	}
+	node3 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "node-3",
+			ResourceVersion: "0",
+		},
+		Status: v1.NodeStatus{
+			Addresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "127.0.0.1",
+				},
+			},
+		},
+	}
+	agentInfo1 = &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-agent-1",
+		},
+		APIPort: 0,
+		PodRef: v1.ObjectReference{
+			ResourceVersion: "0",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-1",
+		},
+	}
+	agentInfo2 = &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-agent-2",
+		},
+		APIPort: 0,
+		PodRef: v1.ObjectReference{
+			ResourceVersion: "0",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-3",
+		},
+	}
+	nameList = []string{"node-1", "node-3"}
+)
+
+func TestLocalSupportBundleRequest(t *testing.T) {
+	getRestClient = func(cmd *cobra.Command) (*rest.RESTClient, error) {
+		supportBundle := &systemv1beta1.SupportBundle{
+			ObjectMeta: metav1.ObjectMeta{Name: "agent"},
+			Status:     systemv1beta1.SupportBundleStatusCollected,
+		}
+		resp, err := json.Marshal(supportBundle)
+		require.NoError(t, err)
+		restClient, _ := rest.RESTClientFor(clientConfig)
```

Comment:
shouldn't we be using https://pkg.go.dev/k8s.io/client-go/rest/fake#RESTClient here?

Reply:
I have used the fake package for restClient.client in the next line

Reply:
My point is that you can simplify this code: I am not sure you need `clientConfig` here. You could use:
```
restClient := fakerest.RESTClient{
    Client: fakerest.CreateHTTPClient(...),
}
```
why do you need a "real" REST Client?
(you may need to replace `*rest.RESTClient` with `rest.Interface` in a few places in the command.go code, but this is a good change anyway).